controllers/nodehealthcheck_controller.go

```diff
@@ -180,6 +183,25 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		// requeue for checking back if template exists later
 		result.RequeueAfter = 15 * time.Second
 		return result, nil
+	} else if template.GetKind() == metal3RemediationTemplateKind {
```

Comment:
Nit: maybe simplify the code to reduce some nesting.

Instead of:
```
else if  template.GetKind()  ..
   if ....
      if ...
```

extract condition into a method:
```
if isSomeGoodMethodName(...)
```

Reply:
Nesting is already reduced in a follow up. IMHO an extra method isn't needed then. 
See https://github.com/slintes/node-healthcheck-operator/blob/292d2b8d042631cfe58483be5771165024eae32f/controllers/resources/templates.go#L79